File: medico/Domain/Medico.Domain/Models/AlcoholHistory.cs
using System;

namespace Medico.Domain.Models
{
    public class AlcoholHistory : Entity
    {
        public string Status { get; set; }

        public string Type { get; set; }

        public int? Amount { get; set; }

        public string Use { get; set; }

        public string Frequency { get; set; }

        public int? Length { get; set; }

        public string Duration { get; set; }

        public bool? Quit { get; set; }

        public int? StatusLength { get; set; }

        public string Notes { get; set; }

        public DateTime? CreateDate { get; set; }

        public Guid PatientId { get; set; }

        public Patient Patient { get; set; }

        public string StatusLengthType { get; set; }
    }
}

File: client/src/dataModels/alcoholHistory.ts
import { TobaccoHistory } from "./TobaccoHistory";
import { DateConverter } from "../helpers/dateConverter";

export class AlcoholHistory extends TobaccoHistory {

    constructor(id: string = "", isDelete: boolean = false,
        patientId: string = "") {

        super(id, isDelete, patientId);
    }

    createFromEntityModel(entity: any) {
        const alcoholHistory =
            new AlcoholHistory(entity.Id, !!entity.isDelete);

        alcoholHistory.StatusLengthType = entity.StatusLengthType;
        alcoholHistory.Status = entity.Status;
        alcoholHistory.Type = entity.Type;
        alcoholHistory.Amount = entity.Amount;
        alcoholHistory.Use = entity.Use;
        alcoholHistory.Frequency = entity.Frequency;
        alcoholHistory.Length = entity.Length;
        alcoholHistory.Duration = entity.Duration;
        alcoholHistory.Quit = !!entity.Quit;
        alcoholHistory.StatusLength = entity.StatusLength;
        alcoholHistory.Notes = entity.Notes;
        alcoholHistory.PatientId = entity.PatientId;
        alcoholHistory.CreateDate = DateConverter
            .sqlServerUtcDateToLocalJsDate(entity.CreateDate);

        return alcoholHistory;
    }
}